shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/scenario/rulealtered/RuleAlteredJobPreparer.java

```diff
@@ -91,29 +91,30 @@ private void prepareTarget(final JobConfiguration jobConfig, final PipelineDataS
             log.info("dataSourcePreparer null, ignore prepare target");
             return;
         }
-        LockContext lockContext = PipelineContext.getContextManager().getInstanceContext().getLockContext();
-        ShardingSphereLock lock = lockContext.getOrCreateSchemaLock(jobConfig.getWorkflowConfig().getSchemaName());
+        final PipelineSimpleLock lock = PipelineSimpleLock.getInstance();
         // TODO make sure only the first thread execute prepare
-        boolean skipPrepare = !lock.tryLock("prepareTargetTablesLock", 100);
+        boolean skipPrepare = !lock.tryLock(PREPARE_LOCK_KEY + jobConfig.getHandleConfig().getJobId(), 100);
         if (skipPrepare) {
-            // TODO polling until first thread finish prepare, then just return
-            while (lock.isLocked("prepareTargetTablesLock")) {
+            while (true) {
                 try {
                     TimeUnit.SECONDS.sleep(1);
                 } catch (InterruptedException e) {
```

Comment:
It's better to limit while loop count, but not `while (true)`, it might cause issue.

Reply:
done